staging/src/k8s.io/component-base/metrics/weighted_histogram.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+
+	"github.com/blang/semver/v4"
+	promext "k8s.io/component-base/metrics/prometheusextension"
+)
+
+// WeightedHistogram is our internal representation for our wrapping struct around prometheus
+// histograms. It implements both kubeCollector and ObserverMetric
+type WeightedHistogram struct {
```

Comment:
Do we have a usecase for using WeightedHistogram in k8s codebase?
My understanding is that for now we don't have any - and if that's true, let's not expose that for now (and treat it purely as an implementation detail for TimingHistogram from k8s perspective).

[less code = less maintenance]

Reply:
done